glean-core/src/metrics/datetime.rs

```diff
@@ -201,33 +203,36 @@ impl DatetimeMetric {
         // try to truncate with `get_iso_time_string` and then parse it back
         // in a `Datetime`. So we need to truncate manually.
         let time = d.time();
-        match tu {
-            TimeUnit::Nanosecond => d.date().and_hms_nano_opt(
+        let dt = match tu {
+            TimeUnit::Nanosecond => d.date_naive().and_hms_nano_opt(
                 time.hour(),
                 time.minute(),
                 time.second(),
                 time.nanosecond(),
             ),
-            TimeUnit::Microsecond => d.date().and_hms_nano_opt(
+            TimeUnit::Microsecond => d.date_naive().and_hms_nano_opt(
                 time.hour(),
                 time.minute(),
                 time.second(),
                 time.nanosecond() / 1000,
             ),
-            TimeUnit::Millisecond => d.date().and_hms_nano_opt(
+            TimeUnit::Millisecond => d.date_naive().and_hms_nano_opt(
                 time.hour(),
                 time.minute(),
                 time.second(),
                 time.nanosecond() / 1000000,
             ),
             TimeUnit::Second => {
-                d.date()
+                d.date_naive()
                     .and_hms_nano_opt(time.hour(), time.minute(), time.second(), 0)
             }
-            TimeUnit::Minute => d.date().and_hms_nano_opt(time.hour(), time.minute(), 0, 0),
-            TimeUnit::Hour => d.date().and_hms_nano_opt(time.hour(), 0, 0, 0),
-            TimeUnit::Day => d.date().and_hms_nano_opt(0, 0, 0, 0),
-        }
+            TimeUnit::Minute => d
+                .date_naive()
+                .and_hms_nano_opt(time.hour(), time.minute(), 0, 0),
+            TimeUnit::Hour => d.date_naive().and_hms_nano_opt(time.hour(), 0, 0, 0),
+            TimeUnit::Day => d.date_naive().and_hms_nano_opt(0, 0, 0, 0),
+        };
+        dt.map(|dt| Utc.from_utc_datetime(&dt).into())
```

Comment:
This appears to be incorrect (Python test failures reinforce this, I think). `d` will have the timezone offset in it.

Reply:
I tried a few things, but I think I'm not understanding the logic here. Would you have a hint for a fix?